clippy_lints/src/literal_representation.rs

```diff
@@ -135,18 +155,25 @@ impl<'a> DigitInfo<'a> {
             (Some(p), s)
         };
 
+        let len = sans_prefix.len();
         let mut last_d = '\0';
         for (d_idx, d) in sans_prefix.char_indices() {
-            if !float && (d == 'i' || d == 'u') || float && (d == 'f' || d == 'e' || d == 'E') {
-                let suffix_start = if last_d == '_' { d_idx - 1 } else { d_idx };
-                let (digits, suffix) = sans_prefix.split_at(suffix_start);
-                return Self {
-                    digits,
-                    radix,
-                    prefix,
-                    suffix: Some(suffix),
-                    float,
-                };
+            let suffix_start = if last_d == '_' {
+                d_idx - 1
+            } else {
+                d_idx
+            };
+            if !float && (d == 'i' || d == 'u') ||
+                float && (d == 'f' || d == 'e' || d == 'E') ||
+                !float && is_possible_suffix_index(&sans_prefix, suffix_start, len) {
```

Comment:
`cargo clippy --all-targets --all-features -- -D clippy::all` fails on this:
```rust
if !float && A || !float && B
```
can be simplified to
```rust
if !float && (A || B)
```
https://travis-ci.org/rust-lang-nursery/rust-clippy/jobs/425293764#L1349-L1365

Everything else LGTM! I'm ready to merge this, after this is fixed.

Unrelated: I think the suggestion of the `nonminimal-bool` lint is wrong or at least way to complicated :thinking:

Reply:
We might be able to tune the minimization algorithm. I'll open an issue